ingest/ledgerbackend/ledger_store.go

```diff
@@ -0,0 +1,54 @@
+package ledgerbackend
+
+import (
+	sq "github.com/Masterminds/squirrel"
+	"github.com/stellar/go/support/db"
+)
+
+// Ledger contains information about a ledger (sequence number and hash)
+type Ledger struct {
+	Sequence uint32 `db:"sequence"`
+	Hash     string `db:"ledger_hash"`
+}
+
+// LedgerStore is used to query ledger data from the Horizon DB
+type LedgerStore interface {
+	// LastLedger returns the highest ledger which is less than `seq` if there exists such a ledger
+	LastLedger(seq uint32) (Ledger, bool, error)
```

Comment:
Still not sure if my suggestion is good but `LastLedger` is a bit confusing.
```suggestion
	LatestLedgerBefore(seq uint32) (Ledger, bool, error)
```